app/views/govuk_publishing_components/components/_document_list.html.erb

```diff
@@ -17,14 +17,28 @@
 
   brand ||= false
   brand_helper = GovukPublishingComponents::AppHelpers::BrandHelper.new(brand)
+
+  li_data_attributes ||= {}
+
+  disable_ga4 ||= false
+  unless disable_ga4
+    ga4_data ||= {}
+    li_data_attributes[:module] = li_data_attributes[:module] ? li_data_attributes[:module] << " ga4-link-tracker" : "ga4-link-tracker"
+    li_data_attributes[:ga4_track_links_only] = ""
+    li_data_attributes[:ga4_link] = {
+      "event_name": "navigation",
+      "type": "document list",
+    }.merge(ga4_data)
+  end
+
 %>
 <% if items.any? %>
   <% unless within_multitype_list %>
     <ul class="<%= classes.join(" ") %>">
   <% end %>
     <% items.each do |item| %>
       <% highlight_class = " gem-c-document-list__item--highlight" if item[:highlight] %>
-      <li class="gem-c-document-list__item<%= within_multitype_list_class %> <%= brand_helper.brand_class %> <%= highlight_class %>">
+      <%= tag.li(class: "gem-c-document-list__item#{within_multitype_list_class} #{brand_helper.brand_class} #{highlight_class}", data: li_data_attributes) do %>
```

Comment:
Using this approach it looks like the GA4 link tracker module would be initialised individually for each list item. Did you look at adding the module to the list instead?

Reply:
@andysellick I decided not to add it to the `<ul>` due to the existence of that `within_multitype_list` boolean. I was wondering if it was set to `true` then the tracking would not exist. Can change it though if you think it should be on the `ul` instead.

Reply:
Looks like we've stumbled into a little rabbit hole here - the `within_multitype_list` option looks like it was added as part of [some other component work](https://github.com/alphagov/govuk_publishing_components/commit/40e9a9d91524c43e5604ae1d18f8b507935ec192) some time ago and was used by another component that wrapped the document list component, which has [since been removed](https://github.com/alphagov/govuk_publishing_components/commit/44e7136052233baf99ef0b4487db83eb84e7e3f5). I suspect this option is no longer in use and should be removed (there's no documentation for it either). That's obviously out of scope for this PR but might be worth raising separately, as another PR or as an issue.

I think if we implement tracking using this approach it would be cleaner to put the module on the parent rather than initialising it multiple times but we should decide the best approach (I'm still thinking about it in light of your responses to my comments).

Reply:
Gone through the rest of your comments - yes, this approach works, if you can put the module on the parent I think that would be better 👍

Reply:
Thanks, I removed `within_multitype_list` in this PR however as the `unless` was interfering with the `<% end %>` statement I needed when using `<%= tag.ul() do %>` - the `<% end %>` for the `unless` statement was closing the `ul` tag instead of being used for the unless statement.

Reply:
Thanks @andysellick - I've removed the extra references. Happy to look into the component wrapper as a separate PR 👍